packages/server/graphql/types/NewMeeting.ts

```diff
@@ -24,6 +24,45 @@ import Team from './Team'
 import TeamMember from './TeamMember'
 import TeamPromptMeeting from './TeamPromptMeeting'
 
+const isMeetingLocked = async (
+  dataLoader: DataLoaderWorker,
+  viewerId: string,
+  teamId: string,
+  endedAt: Date
+) => {
+  const freeLimit = new Date()
+  freeLimit.setDate(freeLimit.getDate() - 30)
+  if (endedAt > freeLimit) {
+    return false
+  }
+  const [team, viewer] = await Promise.all([
+    dataLoader.get('teams').loadNonNull(teamId),
+    dataLoader.get('users').loadNonNull(viewerId)
+  ])
+
+  const {featureFlags} = viewer
+  const {tier, isPaid, orgId} = team
+
+  if (!featureFlags.includes('meetingHistoryLimit')) {
+    return false
+  }
+
+  // because we never update archived teams, archived teams will remain accessible even after downgrading the org.
```

Comment:
+1 it looks like the comment on line 55 is saying the same thing. I'm not sure why the comment would be on this line as the lines below check whether the team is on a valid paying tier. Perhaps it was added by mistake and can be removed?